src/System.Management.Automation/security/Authenticode.cs

```diff
@@ -578,70 +591,81 @@ private static X509Certificate2 GetCertFromChain(IntPtr pSigner)
             DWORD error,
             NativeMethods.WINTRUST_DATA wtd)
         {
+            s_tracer.WriteLine("GetSignatureFromWintrustData: error: {0}", error);
+
             Signature signature = null;
-            X509Certificate2 signerCert = null;
+            IntPtr pProvSigner = IntPtr.Zero;
             X509Certificate2 timestamperCert = null;
+            if (TryGetProviderSigner(wtd.hWVTStateData, ref pProvSigner, ref timestamperCert))
```

Comment:
Same here. Once change to out parameters, you can use the new feature in C# to make the code cleaner.